### DISCUSSION TITLE
Encountered an error while compiling the MOOSE submodule in Molteres

### DISCUSSION THREAD

**Discussion by js-jixu** [idaholab/moose#25240] [Q&A Getting Started] [answered]
Hi, developers.

I installed [Molteres](https://smpark7.github.io/moltres/index.html) on an Ubuntu 20.04 server today.  After successfully downloading all the files, I compiled it using the "make -j8" command. The following errors occurred during the compilation process:

![error](https://github.com/idaholab/moose/assets/99813361/2bf5bc15-478e-4424-bb4d-6e1113ed05b9)

The following are the historical commands from my entire installation process, and I believe I did not make any mistakes:

```
    1  curl -L -O https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-Linux-x86_64.sh
    2  bash Mambaforge-Linux-x86_64.sh -b -p ~/mambaforge3
    3  export PATH=$HOME/mambaforge3/bin:$PATH
    4  mamba init
    5  exit
    6  conda config --add channels https://conda.software.inl.gov/public
    7  mamba install moose-tools moose-libmesh
    8  mkdir ~/projects
    9  cd ~/projects
   10  git clone https://github.com/arfc/moltres.git
   11  cd moltres
   12  git checkout master
   13  git submodule init
   14  git submodule update
   15  conda activate moose
   16  mamba activate moose
   17  mamba create -n moose moose-dev
   18  mamba activate moose
   19  mamba install moose-tools moose-libmesh
   20  make -j8
   21  history
```

I think it may be that MOOSE has made some updates that are not compatible with Moltres. When I separately compile the `/framework` folder in the MOOSE submodule in Moltres, this error also occurs. In addition, I downloaded and compiled MOOSE (excluding the Moltres section) according to the tutorials on the MOOSE official website, and the compilation and testing process of MOOSE was very smooth at this time.

In fact, I successfully installed Molteres and MOOSE in late March. So is there a way for me to use a Moose version from March as a submodule instead of the `[Moose @ af38e9f(https://github.com/arfc/moltres)]` , which is Moltres now using.

The error log is attached.

[error_log.txt](https://github.com/idaholab/moose/files/12374502/error_log.txt)

**Comment by GiudGiud** [comment, marked as answer]
You should try to update the moose submodule in moltres 

go in moltres/moose, then 

git fetch origin
Git checkout origin/devel 

then 
Cd ..
Make

**Reply by js-jixu** [reply]
OK, I'll give it a try now.

**Reply by lindsayad** [reply]
@smpark7 FYI

**Reply by js-jixu** [reply]
> go in moltres/moose, then
> git fetch origin
> git checkout origin/devel

I think this is the proper way to fix my problem. And Moltres can be compiled successfully. Thanks for your help!

**Reply by smpark7** [reply]
Thanks for the ping. Time to update the moose submodule
